File: aat-android/src/main/java/ch/bailu/aat/views/preferences/AbsSolidDialog.java
package ch.bailu.aat.views.preferences;

import android.app.AlertDialog;
import android.content.Context;

import ch.bailu.aat_lib.preferences.AbsSolidType;

public abstract class AbsSolidDialog  {

    public static AlertDialog.Builder createDefaultDialog(Context context, AbsSolidType s) {
        AlertDialog.Builder dialog = new AlertDialog.Builder(context);

        dialog.setTitle(s.getLabel());
        if (s.getIconResource() != 0) {
            dialog.setIcon(s.getIconResource());
        }

        return dialog;
    }


    public static AlertDialog.Builder createDefaultDialog(Context c, String s) {
        AlertDialog.Builder dialog = new AlertDialog.Builder(c);
        dialog.setTitle(s);

        return dialog;
    }

}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/preferences/dialog/AbsSolidDialog.kt
package ch.bailu.aat.views.preferences.dialog

import android.app.AlertDialog
import android.content.Context
import ch.bailu.aat.generated.Images
import ch.bailu.aat_lib.preferences.AbsSolidType

abstract class AbsSolidDialog {
    companion object {
        @JvmStatic
        fun createDefaultDialog(context: Context?, s: AbsSolidType): AlertDialog.Builder {
            val dialog = AlertDialog.Builder(context)
            dialog.setTitle(s.getLabel())
            if ("" != s.getIconResource()) {
                dialog.setIcon(Images.get(s.getIconResource()))
            }
            return dialog
        }
    }
}
